### ISSUE TITLE
FR: Text Tool Ideas

### ISSUE THREAD

**Issue by brimwats** [corphish/TextTools#8]
Hello!

Thank you for all your work on this!

Id love if it was possible to use an open source app like this and not an nearly -adware app like Tooly; these are the text tools that app has, each of which I think would be a good addition to this app:

### Basic Tools

- Text Cases
- Find & Replace
- Sort Lines
- Text Counter
- Repeat Text

### Remove Tools :

- Remove White Spaces
-  Remove Empty Lines
-  Remove Duplicate Lines
-  Remove Duplicate Words
-  Remove Break Lines

### Edit Tools
- Suffix/Prefix
- Add Numbers To Lines
-  Reverse Text
- Reverse Lines
- Reverse Words
- Add Break Lines

**Comment by brimwats** [comment]
A lorem ipsum tool would be neat too!

**Comment by boognish-rising** [comment]
I made the same suggestion re: incorporating the Tooly text tools a while back. The suggestion went by the wayside but that's not to say it wasn't received/considered. I tend to be long winded so I can't get too butthurt when any of my novel-length posts go without a response

_Originally posted by @boognish-rising in https://github.com/corphish/TextTools/issues/3#issuecomment-2161553289_

**Comment by sming** [comment]
Hey, if you want ideas, IMO the closed source Tooly is chock full of great tools, including text tools: https://play.google.com/store/apps/details?id=com.yousx.thetoolsapp

It doesn't add context menu items sadly do beat them to it! 

Cheers

**Comment by corphish** [comment]
Many of the features mentioned were added in the [1.4.0 release](https://github.com/corphish/TextTools/releases/tag/v1.4.0)

Text generator and text counter to be added in upcoming releases.

**Comment by boognish-rising** [comment]
IDK if you'd prefer to keep all user contributed feature requests/ideas that are viewed as an improvement to the app and of benefit to its users. 

Obviously, more critical issues and or sweeping changes deserve their own issue, however where things get a little fuzzy is small minor changes like the one I have in mind currently.

From experience, most devs prefer to keep everything as _decidedly separate_ (probably for their own sanity but also from a feasibility perspective of tackling each issue/[FR] as they come), so I'll use that as my guide, however just LMK if you'd rather they be added to this issue (is any other for that matter).

**Comment by xa4hf8** [comment]
@boognish-rising

I guess it does not matter so much when all of the open issues fit on one page, but the GitHub design intent was that original posts which can be tagged as a bug or feature request belong in the Issues section, and everything else goes into Discussions. Administrators can move posts to Discussions if needed, but the [only post](https://github.com/corphish/TextTools/discussions/19) in Discussions right now is a bug report that should be a closed Issue. 😏

**Comment by boognish-rising** [comment]
>_[...] tagged as a bug or feature request belong in the Issues section, and everything else goes into Discussions. Administrators can move posts to Discussions if needed [...]_

 @xa4hf8  

While I didn't realize it was official, my gut has always been to relay questions and... well. Sidebar _discussions_ on related/relevant topics to the project by creating a new discussion, reserving new issue creation to actual bugs/defects and feature requests only. 

There's a monkey wrench, though. Not all projects have a **Discussions** section (same goes for **Wiki**, iirc. Which makes me wonder if it's an optional thing that perhaps isn't enabled by default 🤔

**Comment by xa4hf8** [comment]
> if it's an optional thing that isn't enabled by default

That's correct - it must be manually activated, and some project managers just don't realize that it exists. You could open an issue to request it after searching the issue database for "discussions" to confirm that it has not already been requested.
